src/skeleton/mod.rs

```diff
@@ -294,18 +295,41 @@ fn extract_cargo_metadata(path: &Path) -> Result<cargo_metadata::Metadata, anyho
 }
 
 /// If the top-level `Cargo.toml` has a `members` field, replace it with
-/// a list consisting of just the specified member.
+/// a list consisting of just the path to the package.
 ///
 /// Also deletes the `default-members` field because it does not play nicely
-/// with a modified `members` field and has no effect on cooking the final receipe.
-fn ignore_all_members_except(manifests: &mut [ParsedManifest], member: String) {
+/// with a modified `members` field and has no effect on cooking the final recipe.
+fn ignore_all_members_except(
+    manifests: &mut [ParsedManifest],
+    metadata: &Metadata,
+    member: String,
+) {
     let workspace_toml = manifests
         .iter_mut()
         .find(|manifest| manifest.relative_path == std::path::PathBuf::from("Cargo.toml"));
 
     if let Some(workspace) = workspace_toml.and_then(|toml| toml.contents.get_mut("workspace")) {
         if let Some(members) = workspace.get_mut("members") {
-            *members = toml::Value::Array(vec![toml::Value::String(member)]);
+            let workspace_root = &metadata.workspace_root;
+            let workspace_packages = metadata.workspace_packages();
+
+            if let Some(pkg) = workspace_packages
+                .into_iter()
+                .find(|pkg| pkg.name == member)
+            {
+                // Make this a relative path to the workspace, and remove the `Cargo.toml` child.
+                // XXX: Do we need to strip the prefix? Or can we use an absolute path?
+                let new_member_path = pkg
+                    .manifest_path
+                    .strip_prefix(workspace_root)
+                    .ok()
+                    .and_then(|mp| mp.parent());
```

Comment:
Let's use https://docs.rs/pathdiff/latest/pathdiff/fn.diff_paths.html here. `pathdiff` is already in the dependency tree.